stdlib/experimental/geo/README.md

```diff
@@ -0,0 +1,201 @@
+It uses golang implementation of S2 Geometry Library [https://s2geometry.io/].
+It is designed to work on a schema with a tags `_ci` which contains S2 cell ID token
+of level decided by the user and fields `lat`, `lon`.
+
+The `_cid` tags hold value of cell ID expressed as token (`s2.CellID.ToToken()`) of corresponding level.
+The S2 cell levels are shown at [https://s2geometry.io/resources/s2cell_statistics.html].
+The level must be decided by the user.
+The rule of thumb is that it should be as high as possible for faster filtering 
+but not too high in order to avoid risk of having high cardinality.
+The token can be easily calculated from lat and lon using Google S2 library which is available for many languages.
+
+The schema may further contain a tag which identifies data source (`id` by default),
+and a field representing track identification (`tid` by default).
+For some use cases a tag denoting point type (with values like `start`/`stop`/`via`, for example) may also be useful.
+
+Examples of line protocol input (`_ci` with cell ID level 11 token):
+```
+taxi,_pt=start,_ci=89c2594 tip=3.75,dist=14.3,lat=40.744614,lon=-73.979424,tid=1572566401123234345i 1572566401947779410
+```
+```
+bike,id=biker-007,_pt=via,_ci=89c25dc lat=40.753944,lon=-73.992035,tid=1572588100i 1572567115
+```
+
+Some functions in this package works on row-wise sets (as it very likely appears in line protocol),
+with fields `lat`, `lon` (and possibly `tid`) as columns.
+That is achieved by correlation by `_time` (and `id` if present) using `pivot()` or provided convenience `geo.toRows()` function.
+Therefore it is advised to store time with nanoseconds precision to avoid false matches in deployments
+where `id` (or any other source identifying) tag is no present.
+
+Fundamental transformations:
+- `gridFilter`
+- `strictFilter`
+- `toRows`
+- `filterRows`
+
+Aggregate operations:
+- `groupByArea`
+- `asTracks`
+
+The package uses the following types:
+- `box` - has the following named float values: `minLat`, `maxLat`, `minLon`, `maxLon`.
+- `circle` - has the following named float values: `lat`, `lon`, `radius`.
+- `polygon` - has `points` value which is an array of objects with named float values `lat` and `lon`.
+
+### Function `gridFilter`
+
+The `gridFilter` filters data by specified box, circle or polygon.
+It calculates grid of tokens that overlays specified region and then uses `_ci` to filter
+against the set.
+The grid cells always overlay the region, therefore result may contain data with latitude and/or longitude outside the region.
+
+This filter function is intended to be fast as it uses `_ci` tag to filter records.
+If precise filtering is needed, `strictFilter()` may be used later (after `toRows()`).
+
+Example:
+```
+from(bucket: "rides")
+  |> range(start: 2019-11-01T00:00:00Z)
+  |> filter(fn: (r) => r._measurement == "taxi")
+  |> geo.gridFilter(box: {minLat: 40.51757813, maxLat: 40.86914063, minLon: -73.65234375, maxLon: -72.94921875})
```

Comment:
@jlapacik The standard library `filter` maintains no state itself, it just calls its lambda for each row it receives.  By implementing a `gridFilter` stream function, it's possible to computing the tiling (a set of S2Cells that cover the region) just one time, and then use it for every row. I like the composability of your proposal, but it seems like it would have some performance impact.

Reply:
@wolffcm Your comment was correct I thought, now I am confused, as of now, there is a call to builtin fn `getGrid` in the `gridFilter` to have precomputed tiling for given region.

@jlapacik I think `gridFilter` cannot be implemented as suggested because at this time of query execution, `lat` and `lon` are in separate streams/tables, this function filters records by tag value.

Reply:
Yes, thinking it over now, you're right, my original comment was right, and I confused myself.  Since the expensive calls to `getFilter` is done outside of the call to `filter(fn: (r) => ...)`, what you have will perform well.